modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/StructuredObjectMarshaller.java

```diff
@@ -166,13 +209,50 @@ private void fillObjectWithReadObjectFrom(
     public void defaultFillFieldsFrom(DataInputStream input, Object object, ClassDescriptor descriptor, UnmarshallingContext context)
             throws IOException, UnmarshalException {
         for (FieldDescriptor fieldDescriptor : descriptor.fields()) {
-            Object fieldValue = valueReader.read(input, context);
-            setFieldValue(object, fieldDescriptor, fieldValue);
+            fillFieldFrom(input, object, context, fieldDescriptor);
         }
     }
 
-    private void setFieldValue(Object target, FieldDescriptor fieldDescriptor, Object value) {
-        fieldDescriptor.accessor().set(target, value);
+    private void fillFieldFrom(DataInputStream input, Object object, UnmarshallingContext context, FieldDescriptor fieldDescriptor)
+            throws IOException, UnmarshalException {
+        if (fieldDescriptor.clazz().isPrimitive()) {
```

Comment:
We can store `isPrimitive` in the field's descriptor, wdyt?

Reply:
`isPrimitive()` seems to be a super-fast computation, so it looks like it does not matter computationally. But if we create a field for it, it would increase code complexity a bit.